### DISCUSSION TITLE
MapController Override Keyboard Shortcuts?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#6203] [Q&A] [answered]
Would there be a way to override the `MapController` default keyboard shortcuts? Currently, the `alt`, `ctrl` and `shift` keys when used alone all perform the same function of allowing the user to rotate/tilt the map.

According to the `MapController/Controller` [docs](https://deck.gl/docs/api-reference/core/controller#options)...

> 
> keyboard (Boolean|Object) - enable interaction with keyboard. Default true. If an object is supplied, it may contain the following fields to customize the keyboard behavior:
> 
>     zoomSpeed (Number) - speed of zoom using +/- keys. Default 2.
>     moveSpeed (Number) - speed of movement using arrow keys, in pixels.
>     rotateSpeedX (Number) - speed of rotation using shift + left/right arrow keys, in degrees. Default 15.
>     rotateSpeedY (Number) - speed of rotation using shift + up/down arrow keys, in degrees. Default 10.

However, there doesn't appear to be a way to change the behavior of the various shortcut keys?

**Comment by Pessimistress** [comment, marked as answer]
Override the [event handler method](https://deck.gl/docs/api-reference/core/controller#handleeventevent).
